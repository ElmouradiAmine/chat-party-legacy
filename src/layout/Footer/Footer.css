.footer {
  color: var(--color-font-primary);
  font-size: 1.2rem;
  opacity: 0.7;

  display: flex;
  align-items: center;
  justify-content: center;
  transition: all 0.3s;
  grid-row: footer-start/footer-end;
  grid-column: full-start/full-end;
}

.footer__copyright {
  margin-right: 1rem;
}
