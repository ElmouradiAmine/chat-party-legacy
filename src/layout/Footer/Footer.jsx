import React from 'react';
import PropTypes from 'prop-types';

import './Footer.css';

const Footer = ({ className }) => (
  <footer className={`footer ${className}`}>
    <div className="footer__copyright">
      Â© 2020 Chat Party, Inc.
    </div>
  </footer>
);

Footer.propTypes = {
  className: PropTypes.string,
};

Footer.defaultProps = {
  className: '',
};

export default Footer;
