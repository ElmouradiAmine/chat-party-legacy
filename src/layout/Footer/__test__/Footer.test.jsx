import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Footer from '../Footer';

afterEach(cleanup);
describe('Footer layout', () => {
  const className = 'className';

  test('renders correctly', () => {
    render(<Footer />);
  });

  test('renders correctly with props', () => {
    const { container } = render(<Footer className={className} />);
    expect(container).not.toBeNull();
    expect(container.firstElementChild?.classList).toContain(className);
  });

  test('contains copyright', () => {
    const { getByText } = render(<Footer />);
    const copyright = getByText('Â© 2020 Chat Party, Inc.');
    expect(copyright).not.toBeNull();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Footer className={className} />);
    expect(tree).toMatchSnapshot();
  });
});
