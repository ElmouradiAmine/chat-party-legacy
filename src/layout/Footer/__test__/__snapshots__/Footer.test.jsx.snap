// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer layout matches snapshot 1`] = `
<footer
  className="footer className"
>
  <div
    className="footer__copyright"
  >
    Â© 2020 Chat Party, Inc.
  </div>
</footer>
`;
